### DISCUSSION TITLE
Delete, with my apologies

### DISCUSSION THREAD

**Discussion by diamondsw** [Homebrew/discussions#282] [Getting started] [answered]
I approached this all wrong and violated the Rule of Wil Wheaton: Don't Be A Dick. I can't delete it (although I'd like to), so I suppose this can just be a reminder not to be an ass and always remember the person on the other side of the computer.

**Comment by SMillerDev** [comment]
The workflow is just the same as for any other processor. Install the script from the homepage and it'll tell you if it encounters issues.

**Comment by carlocab** [comment]
A few things:

1. Perhaps you'd like to open a PR to fix it?
2. M1 Macs are currently unsupported, so it's not unreasonable for them to not show up at all there.
3. The `brew` maintainers have day jobs and lives to live, on top of their never-ending list of things to fix on Homebrew, so your expectations here seem a bit unreasonably high.

**Reply by diamondsw** [reply]
I'm not asking the world. I'm asking that the _front page_ acknowledge that there's been a major change and guide users (which it doesn't). You know, like the documentation promised a month ago should have, but doesn't, because it doesn't exist. So I'm asking for the _absolute barest minimum_. But apparently that's "unreasonably high".

**Reply by SMillerDev** [reply]
You're more than welcome to contribute the bare minimum, because nobody else has time for it. Otherwise you have zero rights to complain.

**Reply by carlocab** [reply]
I'm afraid you just don't understand how free software works. Please read the licenses of the free software that you use. Here's `brew`'s: https://github.com/Homebrew/brew/blob/master/LICENSE.txt

In particular, you should concentrate on the fact that software is provided with no warranty. That means that you don't get to set expectations for the software or its documentation, not even a "bare minimum". It also means that if you think something's broken, the best way to fix that is to fix it yourself and open a PR. Otherwise, you're just hoping someone else sympathises and does the work for you.

**Comment by diamondsw** [comment]
Right now, these are the top links I'm finding. And they're a hodgepodge of out-of-date changing expectations.
https://discourse.brew.sh/t/timeline-for-homebrew-on-arm-macs/8191
https://news.ycombinator.com/item?id=25183120
https://dev.to/peterwitham/brew-and-m1-chip-are-now-friends-5b43
Following that last one finally lands at:
https://github.com/Homebrew/brew/pull/9117
https://github.com/Homebrew/brew/issues/9177
Which is (again) a month old issue with lots of active discussion and changes.

What's going to happen when people get their M1 Macs and upgrade? Who knows? Nothing is documented! Will it just carry everything over? Will it automatically install ARM where it belongs and x86_64 where it belongs, and all work together?

**Comment by diamondsw** [comment]
Oh, I understand how it works. If **that** is the attitude, then just take down the front page entirely and point people to a github page. Because the "we don't support anything we made" attitude is _complete bullshit_ for a user-facing project.

Apparently a couple lines of text and an authoritative link is too much to ask. Instead waste untold numbers of people's time as they upgrade.

**Comment by diamondsw** [comment, marked as answer]
Marking as answer: CLOSED: WONTFIX

**Reply by carlocab** [reply]
This is incorrect. I'm pretty sure you've already been issued invitations by two maintainers to open a PR to try to improve things.

**Comment by iMichka** [comment]
Please be kind to our maintainers, and watch your tone. We are trying to do our best, and do have that much time to work on websites, documentation and stuff: it's really not a priority for us. But anyone with text skills can open a pull request and improve the situation.

> What's going to happen when people get their M1 Macs and upgrade? Who knows? Nothing is documented! Will it just carry everything over? Will it automatically install ARM where it belongs and x86_64 where it belongs, and all work together?

Maybe. All this may take months (years) anyway. So do not expect us to be faster.

**Comment by jonchang** [comment]
Please don't make posts like this. You agreed to follow the [Code of Conduct](https://github.com/Homebrew/.github/blob/296f1a237c003a897cd0887f96cd21ae614f9c20/CODE_OF_CONDUCT.md) when opening discussions here.

> Members of the community are respectful. We're respectful of others, their positions, their skills, their commitments, and their efforts. We're respectful of the volunteer efforts that permeate the Homebrew community. We're respectful of the processes set forth in the community, and we work within them. When we disagree, we are courteous in raising our issues.
